### PR TITLE
Add dyndns test

### PR DESCRIPTION
**Pull request by perry-mitchell**

### CODE REVIEW

**Review by LinusU — APPROVED** [review]
Neat 👍

**Comment by LinusU** [comment]
Hmm, the Travis error is strange, this is the HTML from the site:

```html
<input id="login25_username" name="username" value="">
```

I wonder if they are doing something with `User-Agent` headers or similar...

**edit:** haha 😆 I know what it is!

They aren't setting `type="text"`. The default is `text`, but the query-selector engine doesn't care about that...

Maybe add something like this:

```js
export const USERNAME_QUERIES = [
    "input[type=text][id^=user]",
    "input:not([type])[id^=user]",
    "input[type=text][name^=user]",
    "input:not([type])[name^=user]",
    // ...
];
```

It's a bit annoying to have everything twice though 🤔 

btw. this specific query selector catches it: `input:not([type])[name^=user]`

**Comment by perry-mitchell** [comment]
Ah brilliant! Thank you @LinusU that's perfect!

File: source/inputPatterns.js [hunk 1]
```diff
@@ -23,14 +23,25 @@ export const SUBMIT_BUTTON_QUERIES = [
...
+].reduce(
+    (queries, next) => [
+        ...queries,
+        next.replace(/^input/, "input[type=text]"),
+        next.replace(/^input/, "input:not([type])")
+    ],
+    []
+);
+
 export const USERNAME_QUERIES = [
```
Comment (perry-mitchell) [inline source/inputPatterns.js]: This basically becomes:

```json
[
    "input[type=text][id^=user]",
    "input:not([type])[id^=user]",
    "input[type=text][name^=user]",
    "input:not([type])[name^=user]",
    "input[type=text][id*=username i]",
    "input:not([type])[id*=username i]",
    "input[type=text][title*=username i]",
    "input:not([type])[title*=username i]",
    "input[type=text][placeholder*=username i]",
    "input:not([type])[placeholder*=username i]",
    "input[type=text][name*=email i]",
    "input:not([type])[name*=email i]",
    "input[type=text][name*=login i]",
    "input:not([type])[name*=login i]",
    "input[type=email]",
    "input[aria-label*=username i]",
    "input[aria-label*=email i]"
]
```

**Review by LinusU — APPROVED** [review]
Awesome 🍾

**Comment by perry-mitchell** [comment]
Tried adding tests for Facebook, but their security policies (I'm guessing) don't really allow for me to run anything there using Nightmare.

```json
{
        "name": "Facebook",
        "url": "https://www.facebook.com/",
        "expectedFields": {
            "username": "#login_form input[type=email][name=email]",
            "password": "#login_form input[type=password][name=pass]"
        }
    }
```

**Pull request merged**